google/cloud/pubsub/integration_tests/message_integration_test.cc

```diff
@@ -93,9 +93,10 @@ TEST(MessageIntegrationTest, PublishPullAck) {
       ++i->second;
       return;
     }
-    std::move(h).ack();
     ids.erase(i);
     if (ids.empty()) ids_empty.set_value();
+    lk.unlock();
```

Comment:
Naive observation: It is a little confusing that we go out of our way to drop the lock before `h.ack()` but not before `h.nack()`.

Reply:
Yeah, I think this test will need some cleanup. Filed #4715